app/javascript/src/controllers/media_tag_controller.js

```diff
@@ -5,23 +5,50 @@ import mediaTagTokenWriter from '../modules/media_tag_token_writer.js'
 import buildThumbnail from '../modules/media_thumbnail_builder.js'
 
 export default class extends Controller {
-  static targets = [ "mediaTag", "list"]
+  static targets = [ "mediaTag", "list", "authComponent", "loginButton"]
 
   connect() {
+    // TODO: once we get rid of the legacy media viewer, we shuld remove this from MediaTagComponent
+    //       and remove this line.
+    this.element.querySelectorAll('[data-access-restricted-message]').forEach((elem) => elem.hidden = true)
+
     this.setupThumbnails()
+    this.validateMedia()
+  }
+
+  validateMedia(completeCallback) {
     const validators = this.mediaTagTargets
-      .map((mediaTag) => validator(mediaTag.dataset.authUrl, mediaTagTokenWriter(mediaTag)))
+      .map((mediaTag) => {
+        const mediaContext = { isRestricted: mediaTag.dataset.stanfordOnly === "true" || mediaTag.locationRestricted === "true" }
```

Comment:
Is `locationRestricted` in the `dataset`?

Reply:
Yes! Good eye.